Data store refactor

Nick-Foto opened the pull request.

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+
+
+class DataStoreDirectory(DataStoreABC):
+    store_suffix = None
```
this is an instance attribute since defined in `__init__` so remove from here (plus we have no plan to subclass like I did in first implementation)

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+__email__ = "Gavin.Huttley@anu.edu.au"
+__status__ = "Alpha"
+
+_INCOMPLETE_TABLE = "incomplete"
```
Thinking about this more I think `_NOT_COMPLETED_TABLE` is a better name. The phrase "not completed" indicates something could not be completed. Incomplete implies it's interrupted and might be yet completed. We want the former and not the latter. Refactor to change all occurrences throughout.

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+            raise IOError("member exists")
+
+    @abstractmethod
+    def describe(self):
```
return type should be `cogent3.app.typing.TabularType`

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+
+        return data
+
+    def md5(self, unique_id: str, force=True):
```
return type `str`

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+        return self._members
+
+    @property
+    def logs(self):
```
return type `list[DataMember]`

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+        return self._checksums.get(unique_id, None)
+
+    @property
+    def members(self):
```
return type `list[DataMember]`

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+        )
+
+    @property
+    def incomplete(self):
```
return type `list[DataMember]`

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+        with open_(self.source / _LOG_TABLE / unique_id, mode="w") as out:
+            out.write(data)
+
+    def describe(self):
```
return type should be `cogent3.app.typing.TabularType`

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+    def describe(self):
+        ...
+
+    def open(self, member):
```
Delete, we are only providing `read()`

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+        return open_(self.source / member.unique_id, mode="r")
+
+    @property
+    def summary_logs(self):
```
return type should be `cogent3.app.typing.TabularType`

GavinHuttley commented on src/cogent3/app/data_store_new.py:
```diff
@@ -0,0 +1,340 @@
+        ...
+
+    @property
+    def summary_incomplete(self):
```
return type should be `cogent3.app.typing.TabularType`

GavinHuttley reviewed (changes requested):
Thanks for the hard work! Obviously a few suggestions, but pretty minor.

I'm thinking we want a new `DataStoreABC.completed` property. The `members` property would then be both completed and not completed records.

I still haven't made up my mind about the make / get identifier functions. I'll leave that for the next round.

Nick-Foto commented:
> Thanks for the hard work! Obviously a few suggestions, but pretty minor.
> 
> I'm thinking we want a new `DataStoreABC.completed` property. The `members` property would then be both completed and not completed records.
> 
> I still haven't made up my mind about the make / get identifier functions. I'll leave that for the next round.

I had removed get/make identifier in the previous commit.
Now, I added new property completed.
